playground/utils/video_out.py

```diff
@@ -0,0 +1,216 @@
+                'continuous' - shifts the camera in a cyclic trajectory, determined by fitting a bspline
+                'depth_of_field' - interpolates the depth of field over a static frame (first frame in trajectory)
+            frames_between_cameras (int): How many cameras are inserted between keyframe cameras in the trajectory
+            video_fps (int): FPS of exported video
+            min_dof (float): For 'depth_of_field' mode only, determines the minimum dof for interpolation start / end
+            max_dof (float): For 'depth_of_field' mode only, determines the maximum dof for interpolation start / end
+        """
+        self.renderer = renderer
+
+        # List of kaolin.render.camera.Camera objects forming the trajectory
+        self.trajectory = []
+
+        # Selected interpolation modes
+        self.mode = mode
+
+        # Output path for generated video file
+        self.trajectory_output_path = trajectory_output_path
+
+        # For saving and loading trajectory paths
+        self.cameras_save_path = cameras_save_path
+
+        # Camera FPS -- how many cameras are interpolated between key views
+        self.frames_between_cameras = frames_between_cameras
+
+        # Saved Video FPS
+        self.video_fps = video_fps
+
+        # For depth-of-field interpolation mode only - determines the boundary
+        self.min_dof = min_dof
+        self.max_dof = max_dof
+
+    def add_camera(self, camera: Camera):
+        self.trajectory.append(camera)
+
+    def reset_trajectory(self):
+        self.trajectory = []
+
+    def save_trajectory(self):
+        torch.save(self.trajectory, self.cameras_save_path)
+
+    def load_trajectory(self):
+        self.trajectory = torch.load(self.cameras_save_path)
+
+    def render_dof_trajectory(self):
+        out_video = None
+        old_use_dof = self.renderer.use_depth_of_field
+        old_focus_z = self.renderer.depth_of_field.focus_z
+        try:
+            camera = self.trajectory[0]
+            dofs = np.linspace(self.min_dof, self.max_dof, self.frames_between_cameras)
+            for dof in tqdm(dofs):
+                self.renderer.use_depth_of_field = True
+                self.renderer.depth_of_field.focus_z = dof
+                rgb = self.renderer.render(camera)['rgb']
+
+                if out_video is None:
+                    out_video = cv2.VideoWriter(self.trajectory_output_path, cv2.VideoWriter_fourcc(*'mp4v'),
+                                                self.video_fps, (rgb.shape[2], rgb.shape[1]), True)
+                data = rgb[0].clip(0, 1).detach().cpu().numpy()
+                data = (data * 255).astype(np.uint8)
+                data = cv2.cvtColor(data, cv2.COLOR_BGR2RGB)
+                out_video.write(data)
+        finally:
+            self.renderer.use_depth_of_field = old_use_dof
+            self.renderer.depth_of_field.focus_z = old_focus_z
+
+    def render_linear_trajectory(self, interpolation_mode='polynomial'):
```

Comment:
the "linear" naming seems a little confusing here. What in what sense is polynomial interpolation "linear"? The differences of linear vs. continuous is not obvious to me from the names.